### ISSUE TITLE
FirepowerEnableLogging Auth Token timesout

### ISSUE THREAD

**Issue by Maduin82** [TheAlanNix/cisco-security-tools#1]
FirepowerEnableLogging times out after 30mins (problematic for large rule sets, like my 1300+ line guy).  Looks like the token needs to be refreshed ( as per https://www.cisco.com/c/en/us/td/docs/security/firepower/630/api/REST/Firepower_Management_Center_REST_API_Quick_Start_Guide_630/Connecting_with_a_Client.html)

**Comment by TheAlanNix** [comment]
I just updated this script to refresh the auth token if a 401 is received from the FMC.

Unfortunately, you only get to refresh the token 3 times.  They expire every 30 minutes, so, if the script runs for more than two hours I'll have to try to find another way to tackle this.

I also added some logic to check if all logging is already enabled in the rule.  If so, then the script will skip it.  That should cut down on the time it takes to process large access policies like yours.

Give it a shot and let me know how it goes.  I currently don't have an enormous rule set to test with, so you'll get to be the guinea pig here.
